cfpb/cfgov-refresh
# Prepaid search: Handle product deletions

Updates prepaid product code to handle product deletions. (See [GHE]/CFGOV/prepaid-publication/issues/72 for more information)

## Additions

- `deleted_at` field to PrepaidProduct model
- check for deleted products in `import_prepaid_agreements_data` script
- filter for deleted products in prepaid products search

## Testing

1. Check out branch
2. Run migrations: `python cfgov/manage.py migrate`
3. Run prepaid agreement import, which should mark as deleted the 62 `Sunrise Banks` prepaid products that have been removed from Salesforce: `python cfgov/manage.py runscript import_prepaid_agreements_data`
4. Navigate to [agreements search](http://localhost:8000/data-research/prepaid-accounts/search-agreements/) and verify that it displays 2251 total products (the production version shows 2313 total products)


## Screenshots

### Before
<img width="1180" alt="Screen Shot 2019-11-21 at 2 10 00 PM" src="https://user-images.githubusercontent.com/778171/69382798-d3584480-0c6c-11ea-8e0d-622e528e4158.png">

### After
<img width="1214" alt="Screen Shot 2019-11-21 at 2 16 33 PM" src="https://user-images.githubusercontent.com/778171/69382803-d5ba9e80-0c6c-11ea-97ac-b103885784ef.png">
